src/main/java/net/dv8tion/jda/core/JDABuilder.java

```diff
@@ -448,23 +449,29 @@ public JDABuilder setMaxReconnectDelay(int maxReconnectDelay)
      * <br>The shardId that receives all stuff related to given bot is calculated as follows: shardId == (guildId {@literal >>} 22) % shardTotal;
      * <br><b>PMs are only sent to shard 0.</b>
      *
-     * <p>Please note, that a shard will not even know about guilds not assigned to.
+     * <p>Please note, that a shard will not even know about guilds which are not assigned to it.
      *
      * @param  shardId
      *         The id of this shard (starting at 0).
      * @param  shardTotal
      *         The number of overall shards.
      *
-     * @return Returns the {@link net.dv8tion.jda.core.JDABuilder JDABuilder} instance. Useful for chaining.
+     * @throws net.dv8tion.jda.core.exceptions.AccountTypeException
+     *         If this is used on a JDABuilder for {@link net.dv8tion.jda.core.AccountType#CLIENT AccountType.CLIENT}
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided shard configuration is invalid
+     *         ({@code 0 <= shardId < shardTotal} with {@code shardTotal > 0})
+     *
+     * @return The {@link net.dv8tion.jda.core.JDABuilder JDABuilder} instance. Useful for chaining.
      *
      * @see    net.dv8tion.jda.core.JDA#getShardInfo() JDA.getShardInfo()
      */
     public JDABuilder useSharding(int shardId, int shardTotal)
     {
-        if (shardId < 0 || shardTotal < 2 || shardId >= shardTotal)
-        {
-            throw new RuntimeException("This configuration of shardId and shardTotal is not allowed! 0 <= shardId < shardTotal with shardTotal > 1");
-        }
+        if (accountType != AccountType.BOT)
+            throw new AccountTypeException(AccountType.BOT);
```

Comment:
It does nothing on AccountType.CLIENT and I don't see a reason to allow it then. It can only cause confusion for users.